frontend-react/src/contexts/SessionContext.tsx

```diff
@@ -48,17 +55,40 @@ const SessionProvider = ({
     const isAdminStrictCheck = useMemo(() => {
         return activeMembership?.memberType === MemberType.PRIME_ADMIN;
     }, [activeMembership?.memberType]);
+    const [user, setUser] = useState<UserClaims<CustomUserClaims>>();
+
+    useEffect(() => {
```

Comment:
Rather than making a fetch inside an effect, could we just subscribe to an auth state change?  With the global `OKTA_AUTH` instance getting exported now, we have access to the AuthStateManager:

```tsx
const [authState, setAuthState] = useState(
    OKTA_AUTH.authStateManager.getAuthState()
);

...other stuff

function updateAuthState(_authState: AuthState) {
    setAuthState(_authState);
}

useEffect(() => {
    OKTA_AUTH.authStateManager.subscribe(updateAuthState);

    return () => OKTA_AUTH.authStateManager.unsubscribe(updateAuthState);
}, []);

const context = useMemo(
    () => ({
        oktaToken: authState?.accessToken,
        activeMembership,
        isAdminStrictCheck,
        dispatch,
        initialized: authState !== null && !!initialized,
        user: authState?.idToken?.claims,
    }),
    [
        activeMembership,
        authState,
        dispatch,
        initialized,
        isAdminStrictCheck
    ]
);
```

This should work for logged in --> logged out and logged out --> logged in state changes.  We already have synchronous access to the user claims after an auth state change, so I don't think we should need to make a subsequent request 🤔

Reply:
That asks the question: do we foresee needing OIDC extended user profile information in the future? If we do, we need the getUser call as that hits the OIDC userinfo endpoint. If we don't, we can rely strictly on the claims in the ID token. Right now the userinfo endpoint seems to only be returning sub, email, and email_verified properties. I think only the email_verified claim is not given in the original ID token.

Reply:
That's a great point!  My feeling is that we likely wouldn't need anything other than an email -- at the moment, we only care about the `claims.organization` since a user is acting on behalf of an Organization.  I can't imagine there being any features being specifically around a user rather than an Organization on the website...plus, just skimming through the users on the production Okta, we don't have any of that information anyway 😬 

But ultimately, I think this would be a better question for @chris-kuryak!  Do you foresee our needing to access any extended user information from Okta at some point in the future?  [Here](https://developer.okta.com/docs/reference/api/oidc/#userinfo)'s an example of what kind of information we could get.  However, it seems like we'd have to change our onboarding process to have users fill that extended information in or we'd have to collect it ourselves since it doesn't look like we're doing that now.

Reply:
If we do decide that we need OIDC user profiles, I'll likely update this PR to create a hook that handles the logic of when to call getHook that the SessionProvider component can use.

Reply:
That sounds like a plan to me.  In any case, I don't think this is a blocker -- just a suggestion for simplification if we can afford it!

Reply:
@stephenkao Good question. Yes, currently, I think you are correct that email is all we need for Okta. However, know we do collect more information on our users in SmartSheet, which is essentially our CRM. And in there, we have information around name, location, etc. So I am unsure if we'd need to keep the information in both places, and I think this actually may be a question best answered with O+O team's input.

However, let me ask a couple follow-up questions:
1) How permanent of a decision is this? If we decide we don't need to access the other user information, is this a change we could make in the future to start accessing more of the user info stored in Okta.
2) And when you say "access" the user information (like name, location, etc), what would be potential purposes/uses for doing that through Okta? Is that so we could say, perhaps, display the persons first name on their Dashboard like "Welcome Chris from Dell!"

Reply:
1. Definitely not a permanent decision on the code side, but we should start collecting data in Okta during onboarding as soon as possible if there's a potential we'll want to use it down the road.  The code here is just making a request for whatever data we have stored on Okta users so if we don't collect that data at the start, we'd just fetch empty values.
2. Yep, exactly -- right now, we're just using email, but there's potential to use other data like timezone, address, and phone number to inform future features of the site.
3. Just one HTTP request to the Okta API, which isn't bad.  I was just thinking we could simplify it since we already have the email address available without making the request, but not a problem if we want to keep it!

I think we can go forward with keeping the request -- no harm in keeping it for now :)

Reply:
@stephenkao 
Actually, I would lean the opposite direction for now. Let's keep it simple until we have confirmation we need it. Any harm in that? Especially if we could add it back in later.

I know we don't currently ask for extra information on Okta account creation, but I know Sharon has it in SmartSheet, and I think we could manually add it if we wanted to. HOWEVER, I don't see any point in putting that in now without confirmation.

So I say simplify for now and I will bring it up at next O+O/Exp sync. Sound good?

Reply:
For posterity, we chatted about this in standup and are okay with removing the fetch for now!